src/routes/signup.rs

```diff
@@ -0,0 +1,43 @@
+use actix_web::{web::Json, HttpResponse, ResponseError};
+use serde::Deserialize;
+
+use super::Inject;
+use crate::usecases::signup::{SignUp, SignUpError, SignUpInput};
+
+pub async fn signup(
+    usecase: Inject<dyn SignUp>,
+    json: Json<SignUpPayload>,
+) -> Result<HttpResponse, SignUpError> {
+    usecase
+        .signup(json.into_inner().into())
+        .await
+        .map(|user| HttpResponse::Ok().json(user))
```

Comment:
```suggestion
        .map(HttpResponse::Ok().json)
```
Rustの文法知らないマンなので教えて欲しいのですが、こういう書き方ってできるんですかね

Reply:
できないはず

Reply:
通常の関数ならできるんですが，こいつは self をとる UFCS なので無理だったはず